src/OrchardCore.Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -119,5 +124,29 @@ private string GetPattern(AutoroutePart part)
 
             return pattern;
         }
+
+        private string GenerateUniquePath(string path, AutoroutePart context)
```

Comment:
@sebastienros I think we won't agree on this one :-) ... From my experience a lot of users don't have a clue about principles behind websites and they still need or have to manage a website. They easily gets confused when something unexpected happens, so why not eliminate such things if it is possible. Plus other CMS do the same, e.g. Wordpress never generates a duplicit URL for me.

But I can always register a custom AutoRoute handler in my project so I am ok with removing it from the PR. Can you elaborate a little more on this comment?

> If you really need to externalize the validation logic, you can just create another driver, it doesn't need its own service.

I am not sure if I understand what do you have in mind ... If I remove URL generation from the handler I will need to externalize the validation logic and use it in the handler and in the driver.